packages/dashboard/src/components/tasks/task-inspector.tsx

```diff
@@ -0,0 +1,243 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Grid,
+  Typography,
+} from '@mui/material';
+import { CardContent, useTheme } from '@mui/material';
+import { TaskEventLog, TaskState } from 'api-client';
+import React from 'react';
+import { of, switchMap } from 'rxjs';
+import { AppControllerContext } from '../app-contexts';
+import { AppEvents } from '../app-events';
+import { RmfAppContext } from '../rmf-app';
+import { TaskInfo } from 'react-components';
+import { UserProfileContext } from 'rmf-auth';
+import { Enforcer } from '../permissions';
+import { TaskLogs } from './task-logs';
+
+export interface TableDataGridState {
+  task: TaskState | null;
+  open: boolean;
+  onClose: () => void;
+}
+
+export function TaskInspector({ task, open, onClose }: TableDataGridState): JSX.Element {
+  const theme = useTheme();
+  const rmf = React.useContext(RmfAppContext);
+  const appController = React.useContext(AppControllerContext);
+
+  const [taskState, setTaskState] = React.useState<TaskState | null>(null);
+  const [taskLogs, setTaskLogs] = React.useState<TaskEventLog | null>(null);
+  const [waypoint, setWaypoint] = React.useState({
```

Comment:
I think we should separate this into `startWaypoint` and `finishWaypoint`, since we are not encapsulating that much information in this object